src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/UpdateWorkitemProcedureStepState.sql

```diff
@@ -1,80 +1,81 @@
 **************************************************************/
 --
 -- STORED PROCEDURE
 --     UpdateWorkitemProcedureStepState
 --
 -- DESCRIPTION
---     Adds a UPS-RS workitem.
+--     Updates a UPS-RS workitem procedure step state.
 --
 -- PARAMETERS
---     @partitionKey
---         * The system identifier of the data partition.
---     @workitemUid
---         * The workitem UID.
+--     @workitemKey
+--         * The Workitem Key.
 --     @procedureStepStateTagPath
 --          * The Procedure Step State Tag Path
 --     @procedureStepState
 --          * The New Procedure Step State Value
---     @status
---         * The Workitem status, Either 1(ReadWrite) or 2(Read)
+--     @watermark
+--          * The Workitem Watermark
+--     @proposedWatermark
+--          * The Proposed Watermark for the Workitem
 --
 -- RETURN VALUE
---     The WorkitemKey
+--     The Number of records affected in ExtendedQueryTagString table
 --
 -- EXAMPLE
 --
 /*
 
         BEGIN
-	        DECLARE @partitionKey int = 1
-	        DECLARE @workitemUid varchar(64) = '1.2.840.10008.1.2.184951'
+	        DECLARE @workitemKey BIGINT = 1
 	        DECLARE @procedureStepStateTagPath varchar(64) = '00741000'
 	        DECLARE @procedureStepState varchar(64) = 'SCHEDULED'
-	        DECLARE @status tinyint = 1
+	        DECLARE @watermark BIGINT = 101
+            DECLARE @proposedWatermark BIGINT = 201
 
 	        EXECUTE [dbo].[UpdateWorkitemProcedureStepState] 
-	           @partitionKey
-	          ,@workitemUid
+	           @workitemKey
 	          ,@procedureStepStateTagPath
 	          ,@procedureStepState
-	          ,@status
+              ,@watermark
+	          ,@proposedWatermark
         END
         GO
 
 */
 ------------------------------------------------------------------------
 CREATE OR ALTER PROCEDURE dbo.UpdateWorkitemProcedureStepState
-    @partitionKey                   INT,
-    @workitemUid                    VARCHAR(64),
+    @workitemKey                    BIGINT,
     @procedureStepStateTagPath      VARCHAR(64),
     @procedureStepState             VARCHAR(64),
-    @status                         TINYINT
+    @watermark                      BIGINT,
+    @proposedWatermark              BIGINT
 AS
 BEGIN
     SET NOCOUNT ON
 
     SET XACT_ABORT ON
     BEGIN TRANSACTION
 
-    DECLARE @workitemKey BIGINT
-    DECLARE @newWatermark BIGINT
-    DECLARE @currentProcedureStepStateTagValue VARCHAR(64)
+    IF NOT EXISTS (SELECT WorkitemUid FROM Workitem WHERE WorkitemKey = @workitemKey)
+        THROW 50409, 'Workitem does not exist', 1;
+
     DECLARE @currentDate DATETIME2(7) = SYSUTCDATETIME()
 
-    SELECT
-        @workitemKey = WorkitemKey
-    FROM
-        dbo.Workitem
+    -- Update the workitem watermark
+    UPDATE dbo.Workitem
+    SET
+        Watermark = @proposedWatermark
     WHERE
-        PartitionKey = @partitionKey
-        AND WorkitemUid = @workitemUid
+        WorkitemKey = @workitemKey
```

Comment:
We should have a new index for this scenario

Reply:
Wouldn't IXC_Workitem take care of this? Also WorkitemKey is the PK in Workitem table.